assets/src/dashboard/app/api/apiProvider.js

```diff
@@ -62,35 +57,13 @@ export function reshapeStoryObject(editStoryURL) {
   };
 }
 
-export function reshapeTemplateObject({
-  id,
-  title,
-  tags,
-  colors,
-  createdBy,
-  description,
-  pages,
-}) {
-  return {
-    id,
-    title,
-    createdBy,
-    description,
-    status: 'template',
-    modified: moment('2020-04-07'),
-    metadata: [...tags, ...colors].map((value) =>
-      typeof value === 'object' ? { ...value } : { label: value }
-    ),
-    pages,
-    centerTargetAction: `#${APP_ROUTES.TEMPLATE_DETAIL}?id=${id}`,
-    bottomTargetAction: () => {},
-  };
-}
-
 export default function ApiProvider({ children }) {
   const { api, editStoryURL, pluginDir } = useConfig();
   const [stories, setStories] = useState([]);
-  const [templates, setTemplates] = useState([]);
+
+  const { templates, api: templateApi } = useTemplateApi(wpAdapter, {
+    pluginDir,
+  });
```

Comment:
I moved all template api logic into its own hook so keep the apiProvider from getting too big.